Add rails_best_practices to AbleCop (and bump up to version 0.3.0)

dennmart opened the pull request:
This branch sets up the [rails_best_practices](https://github.com/railsbp/rails_best_practices) gem and its [Pronto runner](https://github.com/mmozuras/pronto-rails_best_practices). I also bumped up the version to 0.3.0 because of a few internal changes (and because bumping up gem versions is fun!).

Due to the rails_best_practices gem looking for its configuration file in the `config/` directory of a project, as well as the configuration file not being a dot-file like the other configuration files we copy over, I had to modify the way the Rails generator copies the file. The way it will work now is by setting up a hash constant that will contain the filename we want to copy from the gem as the key, and the destination as its value.

I also added a [CHANGELOG](https://github.com/ableco/gems.able.co/blob/5c4128a63f84b06552a5d89c8de33b13d85fb1d4/ablecop/CHANGELOG.md) in order to keep track of the changes made to the gem in different versions.

---

### Call for help from other developers!

I'm using the default configuration for the rails_best_practices gem. I did a quick run with this configuration on both the Core and PICI repos, and both repos had multiple warnings (Core had 24, PICI had 74). Most of them seemed to be smaller fixes: For example, almost half of the warnings on the Core repo were for [adding database indices](http://rails-bestpractices.com/posts/2010/07/24/always-add-db-index/). But I didn't check deeper into which of these warnings are something we shouldn't be too strict about.

Before merging this, I wanted to ask for input from other devs about which of the defaults we should change for this gem. The default configuration is [here](https://github.com/railsbp/rails_best_practices#custom-configuration) and detailed explanations for each can be found [here](http://rails-bestpractices.com/).

rezart commented:
@ricardotk002 can you help @dennmart testing this on PICI and seeing the type of warnings that are happening?

ricardotk002 commented:
@rezart sure, I'll do a follow up to this.

dennmart commented:
I moved AbleCop into its own repo, and I moved this branch and pull request too: https://github.com/ableco/ablecop/pull/1

Let's continue the discussion over there! :rocket:
